paddle/fluid/framework/details/inplace_op_pass.cc

```diff
@@ -387,22 +516,7 @@ void GraphView::Build(ir::Graph* g) {
   // Because we insert some new created node. Which may have data race between
   // nodes.
   // resolve data harzards depends on the var nodes in right order.
-  ops_ = SortOpLikeDescOrder(*g);
-
-  // 1. track the nodes which reused previous node in Python memory optimize.
-  // these node can not be inplaced, otherwise may generate a circle in graph.
-  std::unordered_set<std::string> all_vars;
-  for (auto& node : g->Nodes()) {
-    if (node->IsVar()) continue;
-    for (auto& out : node->outputs) {
-      if (out->IsCtrlVar() || out->Var() == nullptr) continue;
-      if (all_vars.count(out->Name())) {
-        dup_nodes_.emplace(out->Name());
```

Comment:
have you removed dup_nodes_?

Reply:
> have you removed dup_nodes_?

no, dup_nodes_ are still there, you can check the comments, this part to code is just used to avoid conflict with the python memory_optimize, now the activation inplace and memory_optimize has been disabled, it's safe to remove the code.  Other dup_nodes_ will be appended in the below section of the same file.

Reply:
`dup_nodes_` is confusing. We need to check why circle occurs if we do not skip `dup_nodes_`.